assets/app/scripts/filters/resources.js

```diff
@@ -236,4 +236,40 @@ angular.module('openshiftConsole')
       }
       return ref;
     };
+  })
+  .filter('orderByDisplayName', function(annotationFilter) {
+    function displayName(obj) {
+      if (!obj) {
+        return '';
+      }
+
+      // Prefer the displayName annotation, but fall back to obj.metadata.name
+      // if not set.
+      var displayName = annotationFilter(obj, 'displayName');
```

Comment:
might be worth baking the "fall back to name" behavior into the `displayName` filter... the only place we're not falling back to metadata.name is in project settings where we show the display name and name separately... could make that an optional `fallback` param to the displayName filter